maestro/lifecycle.py

```diff
@@ -108,12 +110,21 @@ def _create_env(self):
         return dict((str(k), str(v)) for k, v in env.items())
 
     def _test(self, container=None):
-        return subprocess.call(self.command, env=self._create_env()) == 0
+        env = self._create_env()
+        if self.envfrom == "env":
+            return subprocess.call(self.command, env=env) == 0
+        elif self.envfrom == 'stdin':
+            p = subprocess.Popen(self.command, stdin=subprocess.PIPE)
+            p.communicate(json.dumps(env).encode('utf-8'))
+            return p.wait() == 0
+        else:
+            raise ValueError(self.envfrom)
```

Comment:
Put a more explicit error message; I was just typing this quickly yesterday.

Reply:
thanks, done